Source/Shared/Storage/DiskStorage.swift

```diff
@@ -81,6 +81,23 @@ public final class DiskStorage: StorageAware {
       completion?()
     }
   }
+  
+  /**
+   Gets information about the cached object.
+   
+   - Parameter key: Unique key to identify the object in the cache
+   */
+  public func objectMetadata(_ key: String) -> ObjectMetadata? {
+
+    do {
+      let attributes = try fileManager.attributesOfItem(atPath: filePath(key))
+      let fileModifiedDate = Expiry.date(attributes[FileAttributeKey.modificationDate] as! Date)
```

Comment:
@zenangst I assume `modificationDate` can't be nil. But in the off chance that it is, if the file is deleted `fileManager.attributesOfItem` is called, I concluded that we shouldn't return a cache object.